services/discord.py

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+'''
+__author__    = 'Bram Hendrickx'
+__copyright__ = 'Copyright 2016 Bram Hendrickx'
+__license__   = """Eclipse Public License - v 1.0 (http://www.eclipse.org/legal/epl-v10.html)"""
+
+Original script by Bram Hendrickx. https://github.com/jpmens/mqttwarn/blob/master/services/ifttt.py
+Modified to work with the autoremote api https://joaoapps.com/autoremote/
+
+'''
+__author__    = 'Michael Brougham'
+__copyright__ = 'Copyright 2018 Michael Brougham'
+__license__   = """Eclipse Public License - v 1.0 (http://www.eclipse.org/legal/epl-v10.html)"""
+
+
+import requests
+
+def plugin(srv, item):
+    ''' expects (URL) in addrs '''
+
+    srv.logging.debug("*** MODULE=%s: service=%s, target=%s", __file__, item.service, item.target)
+
+    try:
+        srv.logging.debug("Sending to discord webhook")
+
+	url = item.addrs[0]
+
+	payload = "------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name=\"content\"\r\n\r\n" + item.message + "\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name=\"username\"\r\n\r\n" + item.topic + "\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW\r\nContent-Disposition: form-data; name=\"tts\"\r\n\r\nfalse\r\n------WebKitFormBoundary7MA4YWxkTrZu0gW--"
```

Comment:
Does discord not support JSON payloads with `Content-type: application/json`? This looks rather brittle to me, particularly if we just interpolate `item.message` and `item.topic` like that ...

Reply:
I will rewrite this as a general webhook service. I couldn't remember what issues I was having when I made this earlier this year.

Reply:
https://github.com/kyb3r/dhooks

I found this which I may use. If so I'll make a general webhook service as well.